main/routines.c

```diff
@@ -545,10 +555,24 @@ extern const char *baseFilename (const char *const filePath)
 	 */
 	for (i = 0  ;  i < strlen (PathDelimiters)  ;  ++i)
 	{
+#ifdef HAVE_MBLEN
+		const char *p;
+		int ml;
+
+		for (p = filePath  ;  *p != '\0'  ;  ++p)
+		{
+			ml = mblen(p, MB_LEN_MAX);
+			if (ml > 1)
+				p += ml - 1;
+			else if (*p == PathDelimiters [i] && p > tail)
+				tail = p;
+		}
+#else
```

Comment:
cp932 have some letters that contains \x5c in trailing byte of DBCS(double byte character set).
So it should be skipped.